### PR TITLE
Auto saving drafts

### PR DESCRIPTION
**Pull request by yaelirub**
Using a debouncer to call the saving method every 500ms (defined in constants)

Fixes # #10069

To test:
*Creating draft post*
1. Go to Blog posts
2. Go to Drafts tab
3. Start writing a blog post
4. Rerun XCode (as if the app quit suddenly)
5. Go to Blog Posts
Expected: Your post exists and saved as local with the latest changes.

*Editing post*
1. Go to Blog posts
2. Choose an exiting draft post
3. tap edit
4. start editing 
5.rerun XCode
Expected: Your posts updates exists as local with the latest changes.

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -266,4 +228,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: d08615d60b668dfbf99e70a4e11e3869879ca70d
 
-COCOAPODS: 1.5.3
+COCOAPODS: 1.3.1
```
Comment (nheagy) [inline Podfile.lock]: You're using an older version of cocoapods. That might work fine, but let's keep the change out of the repo.

File: WordPress/WordPress.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -7151,6 +7156,7 @@
 				7457666E202B558C00F42E40 /* Sources */,
 				7457666F202B558C00F42E40 /* Frameworks */,
 				74576670202B558C00F42E40 /* Resources */,
+				6595B4E51DABEE482E39D7A8 /* [CP] Copy Pods Resources */,
```
Comment (nheagy) [inline WordPress/WordPress.xcodeproj/project.pbxproj]: some more old cocoapods stuff.

**Comment by yaelirub** [comment]
Hey @nheagy , I fixed the crash issue. I was accessing the post managedObjectContext before it was initialized.

I didn't realize we would like to have the same functionality for editing published posts (which makes total sense!) and so now I am using the same debouncing method for editing of a post. I am updating the test plan of this PR to reflect that 

I updated to the latest cocoapod version.

File: WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift [hunk 3]
```diff
@@ -483,6 +484,20 @@ class AztecPostViewController: UIViewController, PostEditor {
...
+        debouncer.callback = { [weak self] in
+            guard let strongSelf = self else {
+                assertionFailure("self was nil while trying to save a post using Debouncer")
+                return
+            }
+            if strongSelf.post.hasLocalChanges() {
+                guard let context = strongSelf.post.managedObjectContext else {
+                    return
+                }
+                ContextManager.sharedInstance().save(context)
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift]: I love that the `!` is gone now!

File: WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift [hunk 4]
```diff
@@ -461,7 +461,8 @@ class AztecPostViewController: UIViewController, PostEditor {
     ///
     private var mediaPreviewHelper: MediaPreviewHelper? = nil
 
-
+    // For autosaving
```
[the remark this answers is not in the document]
Comment (yaelirub) [reply WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift]: Make sense! Will update the PR with your suggestions. Thank you for taking a look!

**Comment by yaelirub** [comment]
Thank you guys! 🙏🏻

**Comment by nheagy** [comment]
Thanks @yaelirub! 🎉

**Pull request merged**